src/core/infer_request.h

```diff
@@ -137,6 +149,30 @@ class InferenceRequest {
         const size_t idx, const void** base, size_t* byte_size,
         TRITONSERVER_MemoryType* memory_type, int64_t* memory_type_id) const;
 
+    // Returns true if data was added for this input using the
+    // AppendDataForDevice function
+    bool HostPolicySpecificData() const
```

Comment:
Move this up till just after IsShapeTensor and rename to HasHostPolicySpecificData.